pyramid_zipkin/request_helper.py

```diff
@@ -165,15 +165,39 @@ def get_binary_annotations(
     :param response: the Pyramid response object
     :returns: binary annotation dict of {str: str}
     """
-    route = request.matched_route.pattern if request.matched_route else ''
 
     annotations = {
+        'http.request.method': request.method,
+        'network.protocol.version': request.http_version,
+        'url.path': request.path,
+        'server.address': request.server_name,
+        'server.port': str(request.server_port),
+        'url.scheme': request.scheme,
         'http.uri': request.path,
         'http.uri.qs': request.path_qs,
-        'http.route': route,
     }
+
+    if request.user_agent:
+        annotations['user_agent.original'] = request.user_agent
+
+    if request.query_string:
+        annotations["url.query"] = request.query_string
+
+    if request.matched_route:
+        annotations['http.route'] = request.matched_route.pattern
+
+    if request.client_addr:
+        annotations['client.address'] = request.client_addr
+
     if response:
-        annotations['response_status_code'] = str(response.status_code)
+        status_code = response.status_code
+        if isinstance(status_code, int):
+            if status_code >= 500:
```

Comment:
can we copy how otel-python does this https://github.com/open-telemetry/opentelemetry-python-contrib/blob/58a6d3a79c2ab33b80786216e8601e880858b9ab/opentelemetry-instrumentation/src/opentelemetry/instrumentation/utils.py#L58 (I think as is, there is no branch to cover 4xx?)

Reply:
By default, the otel collector will have the status code as `UNSET`.
For OTel http `SERVER` spans with 4xx, we can leave the status code as `UNSET`
https://opentelemetry.io/docs/specs/semconv/http/http-spans/#http-server 
<img width="919" alt="Screenshot 2024-06-28 at 14 22 43" src="https://github.com/Yelp/pyramid_zipkin/assets/36337836/3fc7535a-61f0-4848-aa26-69a69956181e">

Reply:
even so, probably worth making it explicit, rather than depending on behavior in the otel collector?

Reply:
updated to explicitly set `Unset` in code